enhancements/operator-conditions.md

```diff
@@ -0,0 +1,180 @@
+---
+title: operator-conditions
+authors:
+  - "@awgreene"
+reviewers:
+  - "@ecordell"
+approvers:
+  - "@kevinrizza"
+creation-date: 2020-08-24
+last-updated: 2020-08-24
+status: implementable
+see-also:
+  - "N/A"  
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# operator-conditions
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The goal of this enhancement is to allow operators managed by the [Operator Lifecycle Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) operator to communicate conditions that OLM would otherwise be unable to infer. In particular, these conditions will be used by OLM to prevent irreversible actions (such as upgrades) when the operator is in a state that would be problematic.
+
+## Motivation
+
+As part of its role in managing the lifecycle of an operator, OLM infers the state of an operator from the state of Kubernetes resources that define the operator. While this sanity-check provides some level of assurance that an operator is in a given state, there are many complex states specific to each operator that OLM is unable to infer on its own. OLM could better handle the lifecycle of operators by providing them the means to communicate complex states. In particular, there are a number of benefits if an operator managed by OLM could communicate when it should not be upgraded. Consider the following scenario:
+
+>An operator installed with automatic updates is currently performing a migration that could result in data loss if OLM upgrades the operator. While this migration process is running, a new version of the operator becomes available. Since the operator was installed with automatic updates, OLM upgrades the operator without allowing the migration process to finish, resulting in unrecoverable data loss. Had OLM given the operator the ability to communicate that it should not be upgraded, no data loss would have occurred.
+
+As shown in the example above, by allowing operators to report complex condition and influence OLM's management, the overall user experience could be improved for both operator authors and OLM users.
+
+### Goals
+
+- Provide operators with the means to explicitly communicate conditions that influences OLM behavior.
+- Allow the list of "OLM Supported Conditions" to grow in the future.
```

Comment:
Isn't it also a goal to provide authors a way to communicate state to admins, even if it doesn't affect OLM behavior directly?

Reply:
This conversation is probably born from the previous enhancement #23. Since operators are writing the the CRD introduced by OLM directly, they could simply append conditions that are not "OLM Supported Conditions" in order to communicate them.

Reply:
Addressed in the latest version of this PR.